### PR TITLE
Dataset and misc tweaks

### PR DESCRIPTION
**Pull request by pradeep-pyro**
Changed how datasets are implemented. 

- Created a `BaseDataset` that abstracts the graph loading functionality. All datasets derive from this class and primarily take care of handling data split, paths etc.
- Dataset samples are now dictionaries and allow storing additional metadata such as filenames. The graph itself is stored in a key called `"graph"`. Additional metadata can be added in the derived dataset classes.

Other miscellaneous tweaks:

- Exposed PyTorch Lightning `Trainer`'s arguments in the CLI.
- Using a random 20% validation split in the FusionGallery dataset. Previously there was no held out validation set.
- Bugfix to center and scale edge UV-grids.
- Minor fix to save one best checkpoint rather than a whole bunch of them.

### CODE REVIEW

File: classification.py [hunk 1]
```diff
@@ -14,32 +14,43 @@
...
+    default=0,
+    help="Number of workers for the dataloader. NOTE: set this to 0 on Windows, any other value leads to poor performance",
+)
 parser.add_argument(
     "--checkpoint",
     type=str,
     default=None,
     help="Checkpoint file to load weights from for testing",
 )
+parser.add_argument(
```
Comment (JoeLambourne) [inline classification.py:31]: Setting the experiment name is good.

My recommendation would be to have sub-dirs inside the experiment directory, perhaps something like this 

```
./results/$experiment_name/$month_day/$hour_min_second
```
It's much easier to find things by date and time than by the version number which lightning generates by default.   If the user doesn't specify an experiment name then the output could be written to 

```
./results/$month_day/$hour_min_second
```

BRepNet writes logs to
```
{opts.log_dir}/{month_day}/{hour_min_second}
```
which gives folders which look like this
```
    logs/0430/123103/checkpoints
    logs/0430/123103/hparams.yaml
    logs/0430/123103/events.out.tfevents.xxxx
```
I think I like the idea of having a user defined `experiment_name` and will add that to BRepNet myself.

It's also nice to echo the log dir to the shell at the start and end of training.

**Review by JoeLambourne — APPROVED** [review]
The refactoring into the base class is really good.

Everything looks sensible to me.

File: classification.py (hunk 1 above)
Comment (pradeep-pyro) [reply classification.py:31]: I incorporated all these suggestions, thank you.

File: uvnet/models.py [hunk 2]
```diff
@@ -148,42 +148,33 @@ def training_step(self, batch, batch_idx):
         inputs.edata["x"] = inputs.edata["x"].permute(0, 2, 1)
         logits = self.model(inputs)
         loss = F.cross_entropy(logits, labels, reduction="mean")
-        self.log("train_loss", loss)
+        self.log("train_loss", loss, on_step=False, on_epoch=True)
```
Comment (pradeep-pyro) [inline uvnet/models.py:151]: Also made some tweaks to log per epoch using lightning features rather than manually.

**Review by karldd — APPROVED** [review]
All looks good to me.

Unrelated to this PR, but I noticed that we don't include the `.seg` files in the Fusion Gallery dataset we provide. They have to downloaded separately and placed into the same folder. It might be better to include them? Either that or add a note in the readme explaining you need to download them separately and where to put them. 

Also I notice this warning:
```
/opt/anaconda3/envs/uv_net/lib/python3.9/site-packages/dgl/base.py:45: DGLWarning: You are loading a graph file saved by old version of dgl.              Please consider saving it again with the current format.
```
And wonder if dgl might drop support for old binary files at some point... I guess we can cross that bridge when we come to it.

**Comment by pradeep-pyro** [comment]
Thanks!

> Unrelated to this PR, but I noticed that we don't include the `.seg` files in the Fusion Gallery dataset we provide. They have to downloaded separately and placed into the same folder. It might be better to include them? Either that or add a note in the readme explaining you need to download them separately and where to put them.

I was thinking about this as well. It will be convenient for users if we include the seg files.

> And wonder if dgl might drop support for old binary files at some point... I guess we can cross that bridge when we come to it.

Exactly my thought.

**Comment by karldd** [comment]
I've updated the Fusion Gallery data download to include the `.seg` files.
